src/c4/gatherv.t.hh

```diff
@@ -166,6 +166,114 @@ void determinate_gatherv(std::vector<T> &outgoing_data,
   return;
 }
 
+//------------------------------------------------------------------------------------------------//
+template <typename T>
+void indeterminate_allgatherv(std::vector<T> &outgoing_data,
```

Comment:
Could you also a comment block here? Sorry, I'm not very familiar with these types of communication patterns.

Reply:
Added full comment blocks to the header `gatherv.hh` and updated the existing ones to (hopefully) be more descriptive.